libs/workspace-manager/src/lib/features/connect-workspace/connect-workspace.component.ts

```diff
@@ -1,36 +1,109 @@
 import { CommonModule } from '@angular/common';
-import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
 import {
-  FormControl,
-  FormGroup,
-  ReactiveFormsModule,
-  Validators,
-} from '@angular/forms';
+  ChangeDetectionStrategy,
+  Component,
+  inject,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
 import { MatButtonModule } from '@angular/material/button';
-import { MatInputModule } from '@angular/material/input';
+import {
+  CURRENT_WORKSPACE_PATH,
+  Directory,
+} from '@angular-cli-gui/shared/data';
+import {
+  BehaviorSubject,
+  combineLatest,
+  of,
+  Subject,
+  switchMap,
+  takeUntil,
+} from 'rxjs';
 
 import { ConnectWorkspaceService } from '../../data-access/connect-workspace.service';
+import { WorkspaceManagerApiService } from '../../data-access/workspace-manager-api.service';
+import { FilesystemNavigatorComponent } from '../../ui';
 
 @Component({
   selector: 'cli-connect-workspace',
   standalone: true,
-  imports: [CommonModule, MatInputModule, MatButtonModule, ReactiveFormsModule],
+  imports: [CommonModule, MatButtonModule, FilesystemNavigatorComponent],
   templateUrl: './connect-workspace.component.html',
   styleUrls: ['./connect-workspace.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ConnectWorkspaceComponent {
+export class ConnectWorkspaceComponent implements OnInit, OnDestroy {
   connectService = inject(ConnectWorkspaceService);
-  form = new FormGroup(
-    {
-      path: new FormControl('', [Validators.required]),
-    },
-    { updateOn: 'submit' }
-  );
+  workspaceManagerApiService = inject(WorkspaceManagerApiService);
+
+  directories$ = new BehaviorSubject<Directory[]>([]);
+  path$ = new BehaviorSubject<string | null>(null);
+  pathSeparator$ = new BehaviorSubject<string | null>(null);
+  isAngularWorkspace$ = new BehaviorSubject<boolean>(false);
+  private destroyed$ = new Subject<boolean>();
+
+  ngOnInit(): void {
+    this.workspaceManagerApiService
+      .getPathSeparator()
+      .subscribe((separator) => {
+        this.pathSeparator$.next(separator);
+        this.initWorkspacePath();
+      });
+
+    combineLatest([this.path$, this.pathSeparator$])
```

Comment:
`directories$` could just be an observable and you can remove the takeUntil
```suggestion
this.directories$ =  combineLatest([this.path$, this.pathSeparator$]).pipe(
        switchMap(([path, separator]) => {
          path && separator && this.setIsAngularWorkspace(path, separator);
          return path
            ? this.workspaceManagerApiService.getDirectoriesInPath(path)
            : of([]);
        }),
      )
```